internal/privi/repo.go

```diff
@@ -104,3 +141,56 @@ func (r *sqliteRepo) getRecord(did string, rkey string) (record, error) {
 
 	return record, nil
 }
+
+type blob struct {
+	Ref      atdata.CIDLink `json:"cid"`
+	MimeType string         `json:"mimetype"`
+	Size     int64          `json:"size"`
+}
+
+func (r *sqliteRepo) uploadBlob(did string, data []byte, mimeType string) (*blob, error) {
```

Comment:
can we just use the api type here instead of enumerating the args?

Reply:
i was thinking the api request would usually include stuff the server needs, but the server could call out to various components to resolve that request. rn it's the same but not necessarily true in the future? but i guess it's fine to do that for now yeah

Reply:
actually leaving this be for now because i don't wanna do the extra formatting in repo, would rather push that into the server